pkgs/test/test/runner/json_reporter_test.dart

```diff
@@ -587,6 +587,70 @@ void customTest(String name, dynamic Function() testFn) => test(name, testFn);
 ''',
           });
     });
+
+    test('groups and tests with custom locations', () {
+      return _expectReport('''
+      group('group 1 inferred', () {
+        setUpAll(() {});
+        test('test 1 inferred', () {});
+        tearDownAll(() {});
+      });
+      group('group 2 custom', location: TestLocation('file:///foo/group', 123, 234), () {
```

Comment:
Maybe these URIs should be package root relative? Or at least support that?

Reply:
I've added test cases for `org-dartlang-app:///` for the filtering, although it's not clear to me what supporting `org-dartlang-app` here would mean for the JSON... I think the clients of the JSON reporters would always expect file paths? If we supported `org-dartlang-app` here, would they need to be translated to real file paths somewhere?

Reply:
Yeah I do think it will mess up some of these use cases - one thing we can try externally is running tests with build_web_compilers (build_runner test -- -p chrome), and see what all it takes to get things working.

Ultimately none of this works today anyways, so its probably ok if we just leave it in a state where it still doesn't work, as long as internally package:test does something sensible and we can leave any extra bits to later.